app/models/course/condition/assessment.rb

```diff
@@ -40,6 +45,10 @@ def self.dependent_class
     Course::Assessment.name
   end
 
+  def self.on_dependent_status_change(submission)
+    resolve_conditional_for(submission.course_user) if submission.graded?
```

Comment:
Will rebase this base on a separate PR that change the satisfied_by? condition for assessment.

Does (3) graded without minimum grade percentage useful? Or can we collapse them into two state.
1) With minimum grade percentage -> graded with grades given
2) Without minimum grade percentage -> submitted is sufficient